academic_observatory_workflows/workflows/pubmed_telescope.py

```diff
@@ -0,0 +1,1623 @@
+        :param snapshot_expiry_days: How long until the backup snapshot (before this release's upserts and deletes) of the Pubmed table exist in BQ.
+        :param max_processes: Max number of parallel processors.
+        """
+
+        self.observatory_api_conn_id = observatory_api_conn_id
+
+        super().__init__(
+            dag_id=dag_id,
+            start_date=start_date,
+            schedule_interval=schedule_interval,
+            catchup=catchup,
+            airflow_conns=[observatory_api_conn_id],
+            queue=queue,
+            tags=[Tag.academic_observatory],
+        )
+
+        # Databse settings
+        self.cloud_workspace = cloud_workspace
+        self.bq_dataset_id = bq_dataset_id
+        self.bq_table_id = bq_table_id
+        self.main_table_id = f"{cloud_workspace.project_id}.{bq_dataset_id}.{bq_table_id}"
+        self.bq_dataset_description = bq_dataset_description
+        self.snapshot_expiry_days = snapshot_expiry_days
+
+        # Metadata on how to pull out and transform the Pubmed data.
+        self.entity_list = {
+            "baseline": PubmedEntity(
+                type="baseline",
+                sub_key="PubmedArticle",
+                set_key="PubmedArticleSet",
+                pmid_location="MedlineCitation",
+                table_name="upserts",
+                table_description="""Pubmed's PubmedArticle table - Includes all the metadata associated with a journal article citation, 
+                both the metadata to describe the published article, i.e. <MedlineCitation>, and additional metadata often pertaining to 
+                the publication's history or processing at NLM, i.e. <PubMedData>.""",
+            ),
+            "upsert": PubmedEntity(
+                type="upsert",
+                sub_key="PubmedArticle",
+                set_key="PubmedArticleSet",
+                pmid_location="MedlineCitation",
+                table_name="upserts",
+                table_description="""PubmedArticle - Includes all the metadata associated with a journal article citation, both the
+                metadata to describe the published article, i.e. <MedlineCitation>, and additional metadata often
+                pertaining to the publication's history or processing at NLM, i.e. <PubMedData>.""",
+            ),
+            "delete": PubmedEntity(
+                type="delete",
+                sub_key="PMID",
+                set_key="DeleteCitation",
+                pmid_location=None,
+                table_name="deletes",
+                table_description="""DeleteCitation - Indicates one or more <PubmedArticle> or <PubmedBookArticle> that have been deleted. 
+                PMIDs in DeleteCitation will typically have been found to be duplicate citations, or citations to content 
+                that was determined to be out-of-scope for PubMed. It is possible that a PMID would appear in DeleteCitation 
```

Comment:
It would be simpler to just set these directly as variables, then it is easier to refer to them throughout the workflow:

```python
        self.baseline = PubmedEntity(
            type="baseline",
            # sub_key="PubmedArticle",
            # set_key="PubmedArticleSet",
            # pmid_location="MedlineCitation",
            table_name="upserts",
            table_description="""Pubmed's PubmedArticle table - Includes all the metadata associated with a journal article citation, both the metadata to describe the published article, i.e. <MedlineCitation>, and additional metadata often pertaining to the publication's history or processing at NLM, i.e. <PubMedData>.""",
        )
        self.upsert = PubmedEntity(
            type="upsert",
            # sub_key="PubmedArticle",
            # set_key="PubmedArticleSet",
            # pmid_location="MedlineCitation",
            table_name="upserts",
            table_description="""PubmedArticle - Includes all the metadata associated with a journal article citation, both the metadata to describe the published article, i.e. <MedlineCitation>, and additional metadata often pertaining to the publication's history or processing at NLM, i.e. <PubMedData>.""",
        )
        self.delete = PubmedEntity(
            type="delete",
            # sub_key="PMID",
            # set_key="DeleteCitation",
            # pmid_location=None,
            table_name="deletes",
            table_description="""DeleteCitation - Indicates one or more <PubmedArticle> or <PubmedBookArticle> that have been deleted. PMIDs in DeleteCitation will typically have been found to be duplicate citations, or citations to content that was determined to be out-of-scope for PubMed. It is possible that a PMID would appear in DeleteCitation without having been distributed in a previous file. This would happen if the creation and deletion of the record take place on the same day.""",
        )
```